v2/api/containerservice/customizations/managed_cluster_extensions.go

```diff
@@ -142,7 +142,7 @@ func (ext *ManagedClusterExtension) PreReconcileCheck(
 	_ context.Context,
 	obj genruntime.MetaObject,
 	_ genruntime.MetaObject,
-	_ kubeclient.Client,
+	_ *resolver.Resolver,
```

Comment:
Consider updating the ADR as well with the change

Reply:
Done. Good catch.